core/execution/common/liquidity_provision.go

```diff
@@ -505,29 +505,35 @@ func (m *MarketLiquidity) AmendLiquidityProvision(
 		}
 	}
 
-	pendingAmendment := m.liquidityEngine.PendingProvisionByPartyID(party)
-	if !m.liquidityEngine.IsLiquidityProvider(party) && pendingAmendment == nil {
+	if !m.liquidityEngine.IsLiquidityProvider(party) {
 		return ErrPartyNotLiquidityProvider
 	}
 
-	lp := m.liquidityEngine.LiquidityProvisionByPartyID(party)
-	if lp == nil && pendingAmendment == nil {
-		return fmt.Errorf("cannot edit liquidity provision from a non liquidity provider party (%v)", party)
+	pendingAmendment := m.liquidityEngine.PendingProvisionByPartyID(party)
+	currentProvision := m.liquidityEngine.LiquidityProvisionByPartyID(party)
+
+	provisionToCopy := currentProvision
+	if currentProvision == nil {
+		if pendingAmendment == nil {
+			return fmt.Errorf("cannot edit liquidity provision from a non liquidity provider party (%v)", party)
```

Comment:
shouldn't we even panic here? that sounds impossible to happen or a bug if it would seeing the call on line 508?

Reply:
perhaps we should yeah